src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -228,7 +228,28 @@ module.exports = function(
   }
 
 
-  function markUpdate(workInProgress) {
+  function markUpdateIfNecessary(workInProgress) {
+    const current = workInProgress.alternate;
+    const instance = workInProgress.stateNode;
+    let shouldMarkUpdate = false;
+    if (current) {
+      // During an update, we might either need to detach a ref
+      // or to call the componentDidUpdate() hook in the commit phase.
+      const currentRef = current.ref;
```

Comment:
The Ref effect gets marked here: https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiberBeginWork.js#L273-L274